metis/bin/metis_client

```diff
@@ -134,11 +134,31 @@ class MetisClient
       next_blob_hash: upload.next_blob_hash
     )
 
+    response = retry_upload(upload_path, upload) if response.code != '200'
+
     raise MetisClient::Error, "Could not upload file #{upload.file_path}\n#{error(response)}" unless response.code == "200"
 
     json_body(response)
   end
 
+  def retry_upload(upload_path, upload)
+    number_attempts = MetisConfig[:max_retries] || 1
+    for retry_number in 1..number_attempts do
+      response = json_post(upload_path,
+        file_size: upload.file_size,
+        action: 'start',
+        next_blob_size: upload.next_blob_size,
+        next_blob_hash: upload.next_blob_hash,
+        reset: true
```

Comment:
NIT:  Can we only call reset explicitly on a 422 response?  If we get a temporary 503, say, from an apache restart or something, it'd suck to have to restart a 200gb file that could have just been retried.

Reply:
The 422 will come from blob_upload, so this would require more state handling.  If it's too much trouble don't worry.

Reply:
That's a good idea. I think I'll have to add the reset-call to blob_upload anyways, since the error doesn't just happen on upload_start.